components/src/Tabs/TabFocusManager.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class TabFocusManager extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      focusedIndex: 0
+    };
+
+    this.handleArrowNavigation = this.handleArrowNavigation.bind(this);
+    this.focusNextTab = this.focusNextTab.bind(this);
+    this.focusPreviousTab = this.focusPreviousTab.bind(this);
+    this.setFocusToTab = this.setFocusToTab.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { focusedIndex } = this.state;
+    if (prevState.focusedIndex !== focusedIndex) {
+      this.updateFocusedTab();
+    }
+  }
+
+  setFocusToTab(index) {
+    this.setState({
+      focusedIndex: index
+    });
+  }
+
+  handleArrowNavigation(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        e.preventDefault();
+        this.focusPreviousTab();
+        break;
+
+      case "ArrowRight":
+        e.preventDefault();
+        this.focusNextTab();
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  focusNextTab() {
+    const { tabRefs } = this.props;
+
+    this.setState(prevState => ({
+      focusedIndex: prevState.focusedIndex === tabRefs.length - 1 ? 0 : prevState.focusedIndex + 1
+    }));
+  }
+
+  focusPreviousTab() {
+    const { tabRefs } = this.props;
+
+    this.setState(prevState => ({
+      focusedIndex: prevState.focusedIndex === 0 ? tabRefs.length - 1 : prevState.focusedIndex - 1
```

Comment:
So sad. In Ruby, the same line as above would work.
But in javascript. `-1 % 5` returns `-1`.
In ruby, `-1 % 5` returns `4`.

`(prevState.focusedIndex - 1 + tabRefs.length) % tabRefs.length` would work, but it's not as nice. :(